gbour gbour@proformatique.com
Index: asterisk-1.8.2-rc1/channels/chan_sip.c
===================================================================
--- asterisk-1.8.2-rc1.orig/channels/chan_sip.c	2010-12-28 14:07:40.388725397 -0500
+++ asterisk-1.8.2-rc1/channels/chan_sip.c	2010-12-28 14:08:01.288724461 -0500
@@ -7579,10 +7579,9 @@
 	
 	ast_copy_string(buf, value, sizeof(buf));
 
-	sip_parse_host(buf, lineno, &username, &portnum, &transport);
-	
-	if ((hostname = strrchr(username, '@'))) {
+  if ((hostname = strrchr(buf, '@'))) {
 		*hostname++ = '\0';
+    username = buf;
 	}
 	
 	if ((secret = strchr(username, ':'))) {
@@ -7597,7 +7596,7 @@
 	}
 
 	if (ast_strlen_zero(username) || ast_strlen_zero(hostname) || ast_strlen_zero(mailbox)) {
-		ast_log(LOG_WARNING, "Format for MWI subscription is user[:secret[:authuser]]@host[:port][/mailbox] at line %d\n", lineno);
+    ast_log(LOG_WARNING, "Format for MWI subscription is user[:secret[:authuser]]@host[:port]/mailbox at line %d\n", lineno);
 		return -1;
 	}
 	
