gbour gbour@proformatique.com
Index: asterisk-1.8.1/channels/chan_sip.c
===================================================================
--- asterisk-1.8.1.orig/channels/chan_sip.c	2010-12-10 11:35:47.441896899 +0100
+++ asterisk-1.8.1/channels/chan_sip.c	2010-12-10 11:40:21.017397490 +0100
@@ -7571,10 +7571,9 @@
 	
 	ast_copy_string(buf, value, sizeof(buf));
 
-	sip_parse_host(buf, lineno, &username, &portnum, &transport);
-	
-	if ((hostname = strrchr(username, '@'))) {
+  if ((hostname = strrchr(buf, '@'))) {
 		*hostname++ = '\0';
+    username = buf;
 	}
 	
 	if ((secret = strchr(username, ':'))) {
@@ -7589,7 +7588,7 @@
 	}
 
 	if (ast_strlen_zero(username) || ast_strlen_zero(hostname) || ast_strlen_zero(mailbox)) {
-		ast_log(LOG_WARNING, "Format for MWI subscription is user[:secret[:authuser]]@host[:port][/mailbox] at line %d\n", lineno);
+    ast_log(LOG_WARNING, "Format for MWI subscription is user[:secret[:authuser]]@host[:port]/mailbox at line %d\n", lineno);
 		return -1;
 	}
 	
