gbour gbour@proformatique.com
Index: asterisk-1.8.4-rc2/channels/chan_sip.c
===================================================================
--- asterisk-1.8.4-rc2.orig/channels/chan_sip.c	2011-02-28 12:52:46.962066828 -0500
+++ asterisk-1.8.4-rc2/channels/chan_sip.c	2011-02-28 12:52:51.262066183 -0500
@@ -25205,6 +25205,16 @@
 	/* Divvy up the items separated by slashes */
 	AST_NONSTANDARD_APP_ARGS(args, tmp, '/');
 
+  if(args.peerorhost == NULL) {
+    dialog_unlink_all(p, TRUE, TRUE);
+    dialog_unref(p, "unref dialog p from mem fail");
+    /* sip_destroy(p); */
+    ast_log(LOG_ERROR, "SIP resource name is not set or empty");
+
+    *cause = AST_CAUSE_CHANNEL_UNACCEPTABLE;
+    return NULL;
+  }
+
 	/* Find at sign - @ */
 	host = strchr(args.peerorhost, '@');
 	if (host) {
