gbour gbour@proformatique.com
Index: asterisk-1.8.3-rc1/channels/chan_sip.c
===================================================================
--- asterisk-1.8.3-rc1.orig/channels/chan_sip.c	2011-01-20 16:10:22.978090494 -0500
+++ asterisk-1.8.3-rc1/channels/chan_sip.c	2011-01-20 16:10:25.848089847 -0500
@@ -25103,6 +25103,16 @@
 	/* Divvy up the items separated by slashes */
 	AST_NONSTANDARD_APP_ARGS(args, tmp, '/');
 
+  if(args.peerorhost == NULL) {
+    dialog_unlink_all(p, TRUE, TRUE);
+    dialog_unref(p, "unref dialog p from mem fail");
+    /* sip_destroy(p); */
+    ast_log(LOG_ERROR, "SIP resource name is not set or empty");
+
+    *cause = AST_CAUSE_CHANNEL_UNACCEPTABLE;
+    return NULL;
+  }
+
 	/* Find at sign - @ */
 	host = strchr(args.peerorhost, '@');
 	if (host) {
