Index: asterisk-1.8.8.0/main/channel.c
===================================================================
--- asterisk-1.8.8.0.orig/main/channel.c	2011-12-16 09:26:44.963264235 -0500
+++ asterisk-1.8.8.0/main/channel.c	2011-12-16 09:27:31.633262647 -0500
@@ -6035,6 +6035,13 @@
 	struct ast_var_t *current, *newvar;
 	const char *varname;
 
+        if ((parent != NULL) && (child != NULL))
+                manager_event(EVENT_FLAG_CALL, "Inherit",
+                              "Parent: %s\r\n"
+                              "Child: %s\r\n",
+                              parent->name,
+                              child->name);
+
 	AST_LIST_TRAVERSE(&parent->varshead, current, entries) {
 		int vartype = 0;
 
