Index: asterisk-1.8.6.0/main/channel.c
===================================================================
--- asterisk-1.8.6.0.orig/main/channel.c	2011-07-30 19:56:29.000000000 -0400
+++ asterisk-1.8.6.0/main/channel.c	2011-09-01 13:20:11.602932361 -0400
@@ -5957,6 +5957,13 @@
 	struct ast_var_t *current, *newvar;
 	const char *varname;
 
+        if ((parent != NULL) && (child != NULL))
+                manager_event(EVENT_FLAG_CALL, "Inherit",
+                              "Parent: %s\r\n"
+                              "Child: %s\r\n",
+                              parent->name,
+                              child->name);
+
 	AST_LIST_TRAVERSE(&parent->varshead, current, entries) {
 		int vartype = 0;
 
