Index: asterisk-1.8.7.2/main/channel.c
===================================================================
--- asterisk-1.8.7.2.orig/main/channel.c	2011-12-14 08:03:29.413263377 -0500
+++ asterisk-1.8.7.2/main/channel.c	2011-12-14 08:04:25.663266802 -0500
@@ -6009,6 +6009,13 @@
 	struct ast_var_t *current, *newvar;
 	const char *varname;
 
+        if ((parent != NULL) && (child != NULL))
+                manager_event(EVENT_FLAG_CALL, "Inherit",
+                              "Parent: %s\r\n"
+                              "Child: %s\r\n",
+                              parent->name,
+                              child->name);
+
 	AST_LIST_TRAVERSE(&parent->varshead, current, entries) {
 		int vartype = 0;
 
